components/AuthChecker/AuthChecker.tsx

```diff
@@ -0,0 +1,18 @@
+import { useSession } from "@/store/useSession";
+import { useRouter } from "next/router";
+
+type Children = {
+  children: React.ReactNode;
+};
+const AuthChecker = ({ children }: Children) => {
+  const router = useRouter();
+  const { user } = useSession((state) => state);
+  const isLoggedIn = user !== null;
+
+  if (isLoggedIn) {
+    void router.push("/");
+  }
+
+  return <>{!isLoggedIn ? children : null}</>;
```

Comment:
Looks good! However, the fragment here is not necessarily needed from the look of things. Please do consider removing it👀